### ISSUE TITLE
Is it possible to read from a dataset directly instead of a tileset?

### ISSUE THREAD

**Issue by DalinarKh** [mapbox/mapbox-unity-sdk#733]
I'm trying to make an app that will allow users to place their own points on the map for others to find, which requires the dataset to be updated regularly. I know how to programmatically upload new features on a dataset and upload it as you would click the 'export' button while editing a dataset on Mapbox, but changes take too long to appear and I'm worried what might happen if a user tries to upload a tileset while another upload/processing job is still unfinished. Instead, is there a way to directly read the points directly from a dataset to place prefabs on the map (similar to the Data Explorer example on Unity) instead of reading from a tileset?

As a side question, I know that there are performance related benefits to reading from a tileset rather than from a dataset but my knowledge on this is pretty vague. If I do manage to read from datasets directly, what consequences will the app face?

TL;DR: The time it takes to upload and process a tileset is too long for a responsive geocaching app. How can I read data directly from my dataset rather than through a tileset?

Thank you!

**Comment by citizen12** [comment]
You can use any data store you want.  Look at the MapBox Conversions. GeoToWorldGlobePosition function for translating geo coordinates to locations in a Unity scene.  Also, check out this [this video](https://www.youtube.com/watch?v=_Idze_NzDu0).  Matthew does a great job of showing lots of Unity/MapBox/AR scenarios with full walk-throughs.

**Comment by DalinarKh** [comment]
@citizen12 Thanks for your response but the issue is it isn't very feasible to request all geo coordinates across the world everytime a player loads up their map when only their nearby coordinates are relevant. And searching for nearby coordinates would be costly because every single coordinate would have to be checked for whether or not they belonged within an area radius. The reason I want to use Mapbox's dataset/tileset features instead is that I'm assuming that searching for nearby coordinates is already somehow optimized in their backend without having to iterate through every single feature point.

**Comment by BergWerkGIS** [comment]
@DalinarKh yes you can read directly from a dataset.

But our SDK doesn't have a C# wrapper for that yet - and no ETA when it will.

For now you'd have to implement your own.
Dataset api docs: https://www.mapbox.com/api-documentation/#datasets

This thread (https://github.com/mapbox/mapbox-unity-sdk/issues/483) might give you some clues on how to start (although it's about inserting into a dataset and not reading from it).

**Comment by DalinarKh** [comment]
@BergWerkGIS Thanks for your response! I noticed the 'list features' query in the api docs can take parameters like limit and start. Where can I look to find all the possible parameters available? I imagine a parameter that can limit returned features to only those that are under a radius can be very helpful, if one exists.

**Comment by BergWerkGIS** [comment]
> I imagine a parameter that can limit returned features to only those that are under a radius can be very helpful, if one exists.

`List features` doesn't do any spatial queries.
It's just paginating through all existing features.

> I noticed the 'list features' query in the api docs can take parameters like limit and start. Where can I look to find all the possible parameters available?

I've never done this myself but according to the docs

> #using pagination to start the listing **after** feature with the id f6d9
https://www.mapbox.com/api-documentation/#list-features

you'd do a 
```
firstTen = https://api.mapbox.com/datasets/v1/{username}/{dataset_id}/features?limit=10
```

then get the `id` of the 10th feature and use that for the second page (and so on):

```
nextTen = https://api.mapbox.com/datasets/v1/{username}/{dataset_id}/features?limit=10&start=<firstTen.features[9].id>
```

**Comment by DalinarKh** [comment]
Awesome. I'll begin experimenting with this and see what I can do. Thanks for the help!
